spec.html

```diff
@@ -35243,8 +35639,20 @@ <h1>Atom</h1>
         <p>The production <emu-grammar>Atom :: CharacterClass</emu-grammar> evaluates as follows:</p>
         <emu-alg>
           1. Evaluate |CharacterClass| to obtain a CharSet _A_ and a Boolean _invert_.
-          1. Return ! CharacterSetMatcher(_A_, _invert_, _direction_).
+          1. If _UnicodeSets_ is *false* or if every element of _A_ consists of a single character, return ! CharacterSetMatcher(_A_, _invert_, _direction_).
+          1. Assert: _invert_ is *false*.
+          1. Let _D_ be the Disjunction containing:
```

Comment:
I can't tell what this is trying to do. Are you... trying to construct a parse node here? There's not really machinery in the spec which says how to do that, and it ends up being kind of incoherent - we assume Parse Nodes have associated source text.

It would be better to be more explicit here.

Reply:
Link to later discussion: https://github.com/tc39/ecma262/pull/2418#discussion_r793851678